roles/st2chatops/tasks/main.yml

```
@@ -82,7 +82,7 @@
 
 - name: Generate authentication token
   command: st2 auth "{{ st2_auth_username }}" -p "{{ st2_auth_password }}" -t
-  when: task_apikey_not_exists is succeeded and task_user_st2_api_key.changed == false
+  when: task_apikey_not_exists is succeeded and task_user_st2_api_key.changed
```

```suggestion
  when: task_apikey_not_exists is succeeded and task_user_st2_api_key is not changed
```
Using `.changed` like this is not the normal way to test this. We should use actual tests. Same with other tasks below where `.changed` should be replaced with ` is changed` or ` is not changed` (depending on the test).

---

Good catch!
That's most probably the reason why CI smoketests are not happy in this PR.